backend/internal/auth/firebase.go

```diff
@@ -20,3 +23,20 @@ func (a FirebaseAuthenticator) VerifyToken(ctx context.Context, token string) (s
 	}
 	return result.UID, true
 }
+
+func (a FirebaseAuthenticator) GenerateToken(ctx context.Context, email string, pwd string) (string, error) {
+    // Populate struct with fields for user creation
+    var u *auth.UserToCreate
+    u = u.Email(email)
+    u = u.Password(pwd)
+
+    // Create User
+    ur, err := a.Client.CreateUser(ctx, u)
```

Comment:
My bad, didn't catch the bad pointer.
Just switching it to value messed up the u.Email and u.Password calls, so I worked around it:
var user auth.UserToCreate
u := &user

Let me know if theres a better way to do this lol

Reply:
Yeah that's fine, there's really no reason to call `.Email()` and `.Password()` on a pointer instead of a value, so you could just change `u` to an `auth.UserToCreate` and only get its address on line 34 with `&u` but it works just the same